### PR TITLE
Tests running inside docker build

### PR DESCRIPTION
**Pull request by AnikaBettge**
To get the test running inside docker build this things are changed:
- start redis server
- add test data
- change configuration for tests
- strds change used data source
- start webhook server

TODO: test with DB are not running in the moment

### CODE REVIEW

File: tests/test_async_mapset_merging.py [hunk 1]
```diff
@@ -135,8 +135,8 @@ def check_remove_test_mapsets(self):
                 rv = self.server.delete(URL_PREFIX + '/locations/nc_spm_08/mapsets/%s' % mapset,
                                         headers=self.admin_auth_header)
                 print(rv.data)
-                self.assertEqual(rv.status_code, 200, "HTML status code is wrong %i" % rv.status_code)
-                self.assertEqual(rv.mimetype, "application/json", "Wrong mimetype %s" % rv.mimetype)
+                self.assertEqual(rv.status_code, 200, "HTML status code is wrong %i %s"%(rv.status_code, mapset))
+                self.assertEqual(rv.mimetype, "application/json", "Wrong mimetype %s"%rv.mimetype)
```
Comment (neteler) [inline tests/test_async_mapset_merging.py:139]:
```suggestion
                self.assertEqual(rv.status_code, 200, "HTML status code is wrong %i %s" % (rv.status_code, mapset))
                self.assertEqual(rv.mimetype, "application/json", "Wrong mimetype %s" % rv.mimetype)
```

File: tests/test_strds_raster_management.py [hunk 2]
```diff
@@ -38,6 +38,14 @@
 __maintainer__ = "Sören Gebbert"
 __email__ = "soerengebbert@googlemail.com"
 
+location = 'nc_spm_08'
+strds_mapset = 'modis_lst'
+strds_url = (URL_PREFIX +
+    '/locations/%(location)s/mapsets/%(mapset)s/strds'
+    % {'location': location, 'mapset': strds_mapset})
+srtds_data = 'LST_Day_monthly'
```
Comment (neteler) [inline tests/test_strds_raster_management.py:46]:
```suggestion
strds_data = 'LST_Day_monthly'
```

File: tests/test_strds_management.py [hunk 3]
```diff
@@ -78,25 +83,25 @@ def test_list_strds_where_2(self):
     #################### INFO #################################################
 
     def test_strds_info(self):
-        rv = self.server.get(URL_PREFIX + '/locations/ECAD/mapsets/PERMANENT/strds/precipitation_1950_2013_yearly_mm',
+        rv = self.server.get(strds_url + '/%s' % srtds_data,
```
Comment (neteler) [inline tests/test_strds_management.py:86]:
```suggestion
        rv = self.server.get(strds_url + '/%s' % strds_data,
```

File: tests/test_strds_raster_management.py [hunk 4]
```diff
@@ -151,39 +161,38 @@ def test_strds_create_register_unregister_1(self):
     #################### LIST RASTER FROM STRDS ###############################
 
     def test_strds_raster_layer_1(self):
-        rv = self.server.get(URL_PREFIX + '/locations/ECAD/mapsets/PERMANENT/strds/precipitation_1950_2013_yearly_mm/raster_layers',
+        rv = self.server.get(strds_url + '/%s/raster_layers' % srtds_data,
```
Comment (neteler) [inline tests/test_strds_raster_management.py:164]:
```suggestion
        rv = self.server.get(strds_url + '/%s/raster_layers' % strds_data,
```

File: tests/test_strds_raster_management.py [hunk 5]
```diff
@@ -151,39 +161,38 @@ def test_strds_create_register_unregister_1(self):
...
         self.assertEqual(rv.mimetype, "application/json", "Wrong mimetype %s" % rv.mimetype)
 
         map_list = json_loads(rv.data)["process_results"]
-        self.assertEqual(len(map_list), 63)
+        self.assertEqual(len(map_list), 24)
 
     def test_strds_raster_layer_2(self):
-        rv = self.server.get(URL_PREFIX + "/locations/ECAD/mapsets/PERMANENT/strds/precipitation_1950_2013_yearly_mm/raster_layers?"
-                          "where=start_time > '2000-01-01'",
+        rv = self.server.get(strds_url + "/%s/raster_layers?where=start_time >= '2016-01-01'" % srtds_data,
```
Comment (neteler) [inline tests/test_strds_raster_management.py:174]:
```suggestion
        rv = self.server.get(strds_url + "/%s/raster_layers?where=start_time >= '2016-01-01'" % strds_data,
```

File: tests/test_strds_raster_renderer.py [hunk 6]
```diff
@@ -40,16 +40,25 @@
 __maintainer__ = "Soeren Gebbert"
 __email__      = "soerengebbert@googlemail.com"
 
+location = 'nc_spm_08'
+strds_mapset = 'modis_lst'
+strds_url = (URL_PREFIX +
+    '/locations/%(location)s/mapsets/%(mapset)s/strds'
+    % {'location': location, 'mapset': strds_mapset})
+srtds_data = 'LST_Day_monthly'
```
Comment (neteler) [inline tests/test_strds_raster_renderer.py:48]:
```suggestion
strds_data = 'LST_Day_monthly'
```

File: tests/test_strds_management.py [hunk 7]
```diff
@@ -37,35 +37,40 @@
 __maintainer__ = "Sören Gebbert"
 __email__ = "soerengebbert@googlemail.com"
 
+location = 'nc_spm_08'
+strds_mapset = 'modis_lst'
+strds_url = (URL_PREFIX +
+    '/locations/%(location)s/mapsets/%(mapset)s/strds'
+    % {'location': location, 'mapset': strds_mapset})
+srtds_data = 'LST_Day_monthly'
```
Comment (neteler) [inline tests/test_strds_management.py:45]:
```suggestion
strds_data = 'LST_Day_monthly'
```

File: tests/test_strds_management.py [hunk 8]
```diff
@@ -78,25 +83,25 @@ def test_list_strds_where_2(self):
...
 
     def test_strds_create_remove(self):
 
         new_mapset = "strds_test"
-        self.create_new_mapset(mapset_name=new_mapset, location_name="ECAD")
+        self.create_new_mapset(mapset_name=new_mapset, location_name=location)
 
         # Create success
-        rv = self.server.post(URL_PREFIX + '/locations/ECAD/mapsets/%s/strds/test_strds' % new_mapset,
+        rv = self.server.post(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds'%(location, new_mapset),
```
Comment (neteler) [inline tests/test_strds_management.py:104]:
```suggestion
        rv = self.server.post(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds' % (location, new_mapset),
```

PEP8

File: tests/test_strds_management.py [hunk 9]
```diff
@@ -107,7 +112,7 @@ def test_strds_create_remove(self):
         self.assertEqual(rv.mimetype, "application/json", "Wrong mimetype %s" % rv.mimetype)
 
         # Create failure since the strds already exists
-        rv = self.server.post(URL_PREFIX + '/locations/ECAD/mapsets/%s/strds/test_strds' % new_mapset,
+        rv = self.server.post(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds'%(location, new_mapset),
```
Comment (neteler) [inline tests/test_strds_management.py:115]:
```suggestion
        rv = self.server.post(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds' % (location, new_mapset),
```

File: tests/test_strds_management.py [hunk 10]
```diff
@@ -117,29 +122,29 @@ def test_strds_create_remove(self):
         self.assertEqual(rv.status_code, 400, "HTML status code is wrong %i" % rv.status_code)
         self.assertEqual(rv.mimetype, "application/json", "Wrong mimetype %s" % rv.mimetype)
         # Read/check information of the new strds
-        rv = self.server.get(URL_PREFIX + '/locations/ECAD/mapsets/%s/strds/test_strds' % new_mapset,
+        rv = self.server.get(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds'%(location, new_mapset),
```
Comment (neteler) [inline tests/test_strds_management.py:125]:
```suggestion
        rv = self.server.get(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds' % (location, new_mapset),
```

File: tests/test_strds_management.py [hunk 11]
```diff
@@ -117,29 +122,29 @@ def test_strds_create_remove(self):
...
         self.assertEqual(rv.status_code, 200, "HTML status code is wrong %i" % rv.status_code)
         self.assertEqual(rv.mimetype, "application/json", "Wrong mimetype %s" % rv.mimetype)
 
         start_time = json_loads(rv.data)["process_results"]["start_time"]
 
-        self.assertEquals(start_time, "'None'")
+        self.assertEqual(start_time, "'None'")
         # Delete the strds
-        rv = self.server.delete(URL_PREFIX + '/locations/ECAD/mapsets/%s/strds/test_strds' % new_mapset,
+        rv = self.server.delete(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds'%(location, new_mapset),
```
Comment (neteler) [inline tests/test_strds_management.py:135]:
```suggestion
        rv = self.server.delete(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds' % (location, new_mapset),
```

File: tests/test_strds_management.py [hunk 12]
```diff
@@ -117,29 +122,29 @@ def test_strds_create_remove(self):
...
                                 headers=self.admin_auth_header)
         print(rv.data)
         self.assertEqual(rv.status_code, 200, "HTML status code is wrong %i" % rv.status_code)
         self.assertEqual(rv.mimetype, "application/json", "Wrong mimetype %s" % rv.mimetype)
         # Try to delete the strds again to produce an error
-        rv = self.server.delete(URL_PREFIX + '/locations/ECAD/mapsets/%s/strds/test_strds' % new_mapset,
+        rv = self.server.delete(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds'%(location, new_mapset),
```
Comment (neteler) [inline tests/test_strds_management.py:141]:
```suggestion
        rv = self.server.delete(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds' % (location, new_mapset),
```

File: tests/test_strds_management.py [hunk 13]
```diff
@@ -117,29 +122,29 @@ def test_strds_create_remove(self):
...
                                 headers=self.admin_auth_header)
         print(rv.data)
         self.assertEqual(rv.status_code, 400, "HTML status code is wrong %i" % rv.status_code)
         self.assertEqual(rv.mimetype, "application/json", "Wrong mimetype %s" % rv.mimetype)
 
-        rv = self.server.get(URL_PREFIX + '/locations/ECAD/mapsets/%s/strds/test_strds' % new_mapset,
+        rv = self.server.get(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds'%(location, new_mapset),
```
Comment (neteler) [inline tests/test_strds_management.py:147]:
```suggestion
        rv = self.server.get(URL_PREFIX + '/locations/%s/mapsets/%s/strds/test_strds' % (location, new_mapset),
```

File: tests/test_strds_raster_management.py [hunk 14]
```diff
@@ -46,7 +54,8 @@ class STRDSTestCase(ActiniaResourceTestCaseBase):
     def test_strds_creation_error(self):
 
         # This must fail, global mapsets are not allowed to modify
-        rv = self.server.post(URL_PREFIX + '/locations/ECAD/mapsets/PERMANENT/strds/test_strds_register',
+        rv = self.server.post(URL_PREFIX + \
+                                '/locations/%(location)s/mapsets/%(mapset)s/strds/test_strds_register'%{'location': location, 'mapset': strds_mapset},
```
Comment (neteler) [inline tests/test_strds_raster_management.py:58]:
```suggestion
                                '/locations/%(location)s/mapsets/%(mapset)s/strds/test_strds_register' % {'location': location, 'mapset': strds_mapset},
```

File: tests/test_strds_management.py (hunk 8 above)
Comment (neteler) [reply tests/test_strds_management.py:104]: (I didn't catch all of them - better search/replace)

File: .dockerignore [hunk 15]
```diff
@@ -7,6 +7,7 @@ docker
 !docker/actinia-core-dev/actinia.cfg
 !docker/actinia-core-prod/actinia.cfg
 !docker/actinia-core-alpine/actinia.cfg
+ !docker/actinia-core-alpine/actinia-test.cfg
```
Comment (mmacata) [inline .dockerignore:10]: indentation is moved

File: tests/test_async_process_validation.py [hunk 16]
```diff
@@ -304,6 +305,7 @@ def test_async_processing_legacy(self):
                                               http_status=200, status="finished")
         self.assertEqual(len(resp["process_results"]), 6)
 
+    # TODO
```
Comment (mmacata) [inline tests/test_async_process_validation.py:308]: Please add a short note, what the TODO is about

**Review by mmacata — APPROVED** [review]
Nice work, thank you. I have only two minor comments, then please merge.

**Review by neteler — APPROVED** [review]
Thanks for your huge effort to get this running.

**Comment by mmacata** [comment]
I will merge now

**Pull request merged**